pkg/accounting/accounting.go

```diff
@@ -380,7 +381,8 @@ func (c *creditAction) Apply() error {
 		if increasedExpectedDebtReduced.Cmp(c.accountingPeer.earlyPayment) > 0 {
 			err = c.accounting.settle(c.peer, c.accountingPeer)
 			if err != nil {
-				c.accounting.logger.Error(err, "failed to settle with credited peer", c.peer)
+				c.accounting.logger.Debug("failed to settle with credited peer", "peer", c.peer, "error", err)
+				c.accounting.logger.Error(nil, "failed to settle with credited peer")
```

Comment:
If you've got an err and you're doing a logger.Error, why not provide err as the first argument instead of nil?     This applies to lots of similar updates throughout the file.   If you're doing the log anyway, why not provide the visibility to the underlying error that triggered the log?